### ISSUE TITLE
Modular application cannot bundle (embed) neo4j. JLink failures

### ISSUE THREAD

**Issue by alwins0n** [neo4j/neo4j#12994]
I tried to embed neo4j into a modularized (java9+) application. Native packaging via jpackage and jlink commands fail due to multiple issues in neo4j jars and dependents. See below

**Neo4j Version:** 5.2.0 / 5.1.0
**Operating System:** Ubuntu
**Driver:**  5.2.0 / 5.10
**JDK:** jdk17

### Steps to reproduce
1. Include neo4j into modularized project (module-info.java)
2. Try to package the application for native distribution via jpackage (sample invocation: /opt/jdk-17.0.1+12/bin/jpackage -n MyApp -p target:target/lib -m com.myapp/MyApp where target/lib contains neo4j with transitive dependencies (e.g. via maven copy-dependencies) 

### Expected behavior
Neo4j should be packageable via jpackage. Each module should have a unique name and no shared packages.

### Actual behavior
Multiple problems 

- split packages (e.g. "jlink failed with: Error: Modules org.neo4j.internal.kernel.api and org.neo4j.cypher.internal.runtime.util export package org.neo4j.internal.kernel.api.helpers to module io.netty.transport")
- wrong Automatic module names (e.g. neo4j-cypher-cache.jar META-INF/Manifest.mf should be Automatic-Module-Name: org.neo4j.cypher.internal.cache instead of .options which is already used by another jar) 
- scala packages without manifest cause problems because of invalid automatic module name derived from filename (this could/should be fixed upstream - e.g. include an automatic module name in jar)
- possibly more. I gave up at some point

**Comment by mnd999** [comment]
Looks to be a duplicate of #12952

**Comment by alwins0n** [comment]
You are right. Additionally, I identified a wrong automatic module name, etc. but its the same problem essentially
